### ISSUE TITLE
Fix first-time build of nanoflann submodule

### ISSUE THREAD

**Issue by jlblancoc** [MRPT/mrpt#1139]
Error, e.g. building in lgtm.com: 

```
[2021-02-20 23:08:04] [build-err] CMake Error at _lgtm_build_dir/3rdparty/nanoflann/nanoflannConfig.cmake:27 (include):
[2021-02-20 23:08:04] [build-err]   include could not find load file:
[2021-02-20 23:08:04] [build-err]     /opt/src/_lgtm_build_dir/3rdparty/nanoflann/nanoflannTargets.cmake
[2021-02-20 23:08:04] [build-err] Call Stack (most recent call first):
[2021-02-20 23:08:04] [build-err]   cmakemodules/script_nanoflann.cmake:6 (find_package)
[2021-02-20 23:08:04] [build-err]   CMakeLists.txt:274 (include)
```

Solution: Invoke cmake during configure step to generate targets.cmake.

**Comment by jlblancoc** [comment]
This seems to only happen on LGTM. Dropping priority.

**Comment by jlblancoc** [comment]
Should be fixed with 8e55362464502cd0755ad225b85fb19d2bc6025c 
Test pending.
